src/Symfony/Component/Translation/Extractor/PhpAstExtractor.php

```diff
@@ -39,7 +40,11 @@ public function __construct(
             throw new \LogicException(sprintf('You cannot use "%s" as the "nikic/php-parser" package is not installed. Try running "composer require nikic/php-parser".', static::class));
         }
 
-        $this->parser = (new ParserFactory())->create(ParserFactory::PREFER_PHP7);
+        if (method_exists(ParserFactory::class, 'createForVersion')) {
+            $this->parser = (new ParserFactory())->createForVersion(PhpVersion::fromString('8.2'));
```

Comment:
why hardcoding PHP 8.2 here ? Shouldn't we parse for the current PHP version instead ?

Reply:
replaced this to make use of `createForHostVersion()` instead